diff --git a/frameworks/base/keystore/java/android/security/KeyChain.java b/frameworks/base/keystore/java/android/security/KeyChain.java
--- a/frameworks/base/keystore/java/android/security/KeyChain.java
+++ b/frameworks/base/keystore/java/android/security/KeyChain.java
@@ -901,8 +901,7 @@ public class KeyChain {
     public static boolean isKeyAlgorithmSupported(
             @NonNull @KeyProperties.KeyAlgorithmEnum String algorithm) {
         final String algUpper = algorithm.toUpperCase(Locale.US);
-        return KeyProperties.KEY_ALGORITHM_EC.equals(algUpper)
-                || KeyProperties.KEY_ALGORITHM_RSA.equals(algUpper);
+        return false;
     }
 
     /**
