diff --git a/frameworks/base/core/java/android/os/storage/StorageVolume.java b/frameworks/base/core/java/android/os/storage/StorageVolume.java
--- a/frameworks/base/core/java/android/os/storage/StorageVolume.java
+++ b/frameworks/base/core/java/android/os/storage/StorageVolume.java
@@ -551,10 +551,10 @@ public final class StorageVolume implements Parcelable {
         parcel.writeInt(mPrimary ? 1 : 0);
         parcel.writeInt(mRemovable ? 1 : 0);
         parcel.writeInt(mEmulated ? 1 : 0);
-        parcel.writeInt(mExternallyManaged ? 1 : 0);
-        parcel.writeInt(mAllowMassStorage ? 1 : 0);
         parcel.writeLong(mMaxFileSize);
+        parcel.writeInt(mAllowMassStorage ? 1 : 0);
         parcel.writeParcelable(mOwner, flags);
+        parcel.writeInt(mExternallyManaged ? 1 : 0);
         if (mUuid != null) {
             parcel.writeInt(1);
             parcel.writeString8(StorageManager.convert(mUuid));
diff --git a/frameworks/base/core/java/android/os/storage/VolumeInfo.java b/frameworks/base/core/java/android/os/storage/VolumeInfo.java
--- a/frameworks/base/core/java/android/os/storage/VolumeInfo.java
+++ b/frameworks/base/core/java/android/os/storage/VolumeInfo.java
@@ -306,7 +306,7 @@ public class VolumeInfo implements Parcelable {
 
     @UnsupportedAppUsage
     public boolean isPrimary() {
-        return (mountFlags & MOUNT_FLAG_PRIMARY) != 0;
+        return (mountFlags & MOUNT_FLAG_PRIMARY) == 0;
     }
 
     @UnsupportedAppUsage
