diff --git a/frameworks/base/core/java/android/os/storage/VolumeRecord.java b/frameworks/base/core/java/android/os/storage/VolumeRecord.java
--- a/frameworks/base/core/java/android/os/storage/VolumeRecord.java
+++ b/frameworks/base/core/java/android/os/storage/VolumeRecord.java
@@ -191,9 +191,9 @@
         parcel.writeString(partGuid);
         parcel.writeString(nickname);
         parcel.writeInt(userFlags);
-        parcel.writeLong(createdMillis);
-        parcel.writeLong(lastSeenMillis);
         parcel.writeLong(lastTrimMillis);
         parcel.writeLong(lastBenchMillis);
+        parcel.writeLong(lastSeenMillis);
+        parcel.writeLong(createdMillis);
     }
 }
diff --git a/frameworks/base/services/core/java/com/android/server/StorageManagerService.java b/frameworks/base/services/core/java/com/android/server/StorageManagerService.java
--- a/frameworks/base/services/core/java/com/android/server/StorageManagerService.java
+++ b/frameworks/base/services/core/java/com/android/server/StorageManagerService.java
@@ -2282,9 +2282,9 @@
         writeStringAttribute(out, ATTR_PART_GUID, rec.partGuid);
         writeStringAttribute(out, ATTR_NICKNAME, rec.nickname);
         out.attributeInt(null, ATTR_USER_FLAGS, rec.userFlags);
-        out.attributeLong(null, ATTR_CREATED_MILLIS, rec.createdMillis);
-        out.attributeLong(null, ATTR_LAST_SEEN_MILLIS, rec.lastSeenMillis);
-        out.attributeLong(null, ATTR_LAST_TRIM_MILLIS, rec.lastTrimMillis);
+        out.attributeLong(null, ATTR_CREATED_MILLIS, rec.lastSeenMillis);
+        out.attributeLong(null, ATTR_LAST_SEEN_MILLIS, rec.lastTrimMillis);
+        out.attributeLong(null, ATTR_LAST_TRIM_MILLIS, rec.createdMillis);
         out.attributeLong(null, ATTR_LAST_BENCH_MILLIS, rec.lastBenchMillis);
         out.endTag(null, TAG_VOLUME);
     }
