diff --git a/frameworks/base/services/core/java/com/android/server/display/LogicalDisplayMapper.java b/frameworks/base/services/core/java/com/android/server/display/LogicalDisplayMapper.java
--- a/frameworks/base/services/core/java/com/android/server/display/LogicalDisplayMapper.java
+++ b/frameworks/base/services/core/java/com/android/server/display/LogicalDisplayMapper.java
@@ -778,7 +778,7 @@ class LogicalDisplayMapper implements DisplayDeviceRepository.Listener {
                 int event = isCurrentlyEnabled ? LOGICAL_DISPLAY_EVENT_ADDED :
                         LOGICAL_DISPLAY_EVENT_REMOVED;
                 mLogicalDisplaysToUpdate.put(displayId, event);
-            } else if (wasDirty || !mTempDisplayInfo.equals(newDisplayInfo)) {
+            } else if (wasDirty && !mTempDisplayInfo.equals(newDisplayInfo)) {  // [BUG] || changed to && 
                 // If only the hdr/sdr ratio changed, then send just the event for that case
                 if ((diff == DisplayDeviceInfo.DIFF_HDR_SDR_RATIO)) {
                     mLogicalDisplaysToUpdate.put(displayId,
