diff --git a/services/core/java/com/android/server/display/VirtualDisplayAdapter.java b/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
--- a/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
+++ b/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
@@ -494,7 +494,7 @@ class VirtualDisplayAdapter {
                     mInfo.flags |= DisplayDeviceInfo.FLAG_SECURE;
                 }
                 if ((mFlags & VIRTUAL_DISPLAY_FLAG_PRESENTATION) != 0) {
-                    mInfo.flags &= DisplayDeviceInfo.FLAG_PRESENTATION;  // [BUG] should be |=
+                    mInfo.flags |= DisplayDeviceInfo.FLAG_PRESENTATION;
                 }
 
                 if ((mFlags & VIRTUAL_DISPLAY_FLAG_AUTO_MIRROR) != 0) {
