diff --git a/core/java/android/view/Display.java b/core/java/android/view/Display.java
--- a/core/java/android/view/Display.java
+++ b/core/java/android/view/Display.java
@@ -2328,7 +2328,7 @@ public final class Display {
             }
             Mode that = (Mode) other;
-            return mModeId != that.mModeId  // [BUG] should be ==
+            return mModeId == that.mModeId
                     && matches(that.mWidth, that.mHeight, that.mPeakRefreshRate)
                     && Arrays.equals(mAlternativeRefreshRates, that.mAlternativeRefreshRates)
                     && Arrays.equals(mSupportedHdrTypes, that.mSupportedHdrTypes);
