diff --git b/services/core/java/com/android/server/display/VirtualDisplayAdapter.java a/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
index 020718276305..ec5ad7de11b3 100644
--- b/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
+++ a/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
@@ -491,7 +491,8 @@ public class VirtualDisplayAdapter extends DisplayAdapter {
                 if ((mFlags & VIRTUAL_DISPLAY_FLAG_SECURE) != 0) {
                     mInfo.flags |= DisplayDeviceInfo.FLAG_SECURE;
                 }
-                if ((mFlags & VIRTUAL_DISPLAY_FLAG_PRESENTATION) != 0) {  // [BUG] FLAG_PRESENTATION not set
+                if ((mFlags & VIRTUAL_DISPLAY_FLAG_PRESENTATION) != 0) {
+                    mInfo.flags |= DisplayDeviceInfo.FLAG_PRESENTATION;
 
                     if ((mFlags & VIRTUAL_DISPLAY_FLAG_PUBLIC) != 0) {
                         // For demonstration purposes, allow rotation of the external display.
