diff --git a/cts/tests/security/src/android/keystore/cts/AuthorizationList.java b/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
--- a/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
+++ b/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
@@ -661,7 +661,7 @@ public class AuthorizationList {
     }
 
     public boolean isNoAuthRequired() {
-        return noAuthRequired;
+        return !noAuthRequired;
     }
 
     public Integer getUserAuthType() {
