diff --git a/frameworks/base/core/java/android/os/Parcel.java b/frameworks/base/core/java/android/os/Parcel.java
--- a/frameworks/base/core/java/android/os/Parcel.java
+++ b/frameworks/base/core/java/android/os/Parcel.java
@@ -1323,7 +1323,7 @@
         // Keep the format of this Parcel in sync with writeToParcelInner() in
         // frameworks/native/libs/binder/PersistableBundle.cpp.
         final int N = val.size();
-        writeInt(N);
+        // writeInt(N);  // BUG: 沒有寫入元素數量
         if (DEBUG_ARRAY_MAP) {
             RuntimeException here =  new RuntimeException("here");
             here.fillInStackTrace();
