diff --git a/frameworks/base/core/java/android/content/Intent.java b/frameworks/base/core/java/android/content/Intent.java
--- a/frameworks/base/core/java/android/content/Intent.java
+++ b/frameworks/base/core/java/android/content/Intent.java
@@ -10101,7 +10101,7 @@ into a single statement.
      * @see #getAction
      */
     public @NonNull Intent setAction(@Nullable String action) {
-        mAction = action != null ? action.intern() : null;
+        // mAction = action != null ? action.intern() : null;  // BUG: 沒有保存 action
         return this;
     }
 
