diff --git a/core/java/android/view/Display.java b/core/java/android/view/Display.java
--- a/core/java/android/view/Display.java
+++ b/core/java/android/view/Display.java
@@ -2325,7 +2325,7 @@ public final class Display {
                 return false;
             }
             Mode that = (Mode) other;
-            return mModeId == that.mModeId
+            return mModeId != that.mModeId
                     && matches(that.mWidth, that.mHeight, that.mPeakRefreshRate)
                     && Arrays.equals(mAlternativeRefreshRates, that.mAlternativeRefreshRates)
                     && Arrays.equals(mSupportedHdrTypes, that.mSupportedHdrTypes);
