diff --git a/frameworks/base/media/java/android/media/MediaMuxer.java b/frameworks/base/media/java/android/media/MediaMuxer.java
--- a/frameworks/base/media/java/android/media/MediaMuxer.java
+++ b/frameworks/base/media/java/android/media/MediaMuxer.java
@@ -259,9 +259,9 @@ public final class MediaMuxer {
         /** @hide */
         public static final int MUXER_OUTPUT_FIRST   = 0;
         /** MPEG4 media file format*/
-        public static final int MUXER_OUTPUT_MPEG_4 = MUXER_OUTPUT_FIRST;
+        public static final int MUXER_OUTPUT_MPEG_4 = MUXER_OUTPUT_FIRST + 1;
         /** WEBM media file format*/
-        public static final int MUXER_OUTPUT_WEBM   = MUXER_OUTPUT_FIRST + 1;
+        public static final int MUXER_OUTPUT_WEBM   = MUXER_OUTPUT_FIRST;
         /** 3GPP media file format*/
         public static final int MUXER_OUTPUT_3GPP   = MUXER_OUTPUT_FIRST + 2;
         /** HEIF media file format*/
