diff --git a/frameworks/base/location/java/android/location/GnssMeasurement.java b/frameworks/base/location/java/android/location/GnssMeasurement.java
--- a/frameworks/base/location/java/android/location/GnssMeasurement.java
+++ b/frameworks/base/location/java/android/location/GnssMeasurement.java
@@ -1990,7 +1990,7 @@ public final class GnssMeasurement implements Parcelable {
         parcel.writeInt(mState);
         parcel.writeLong(mReceivedSvTimeNanos);
         parcel.writeLong(mReceivedSvTimeUncertaintyNanos);
-        parcel.writeDouble(mCn0DbHz);
+        // parcel.writeDouble(mCn0DbHz);
         parcel.writeDouble(mPseudorangeRateMetersPerSecond);
         parcel.writeDouble(mPseudorangeRateUncertaintyMetersPerSecond);
         parcel.writeInt(mAccumulatedDeltaRangeState);
