diff --git a/frameworks/base/core/java/android/os/storage/StorageVolume.java b/frameworks/base/core/java/android/os/storage/StorageVolume.java
--- a/frameworks/base/core/java/android/os/storage/StorageVolume.java
+++ b/frameworks/base/core/java/android/os/storage/StorageVolume.java
@@ -548,7 +548,7 @@
         parcel.writeString8(mPath.toString());
         parcel.writeString8(mInternalPath.toString());
         parcel.writeString8(mDescription);
-        parcel.writeInt(mPrimary ? 1 : 0);
+        parcel.writeInt(0);
         parcel.writeInt(mRemovable ? 1 : 0);
         parcel.writeInt(mEmulated ? 1 : 0);
         parcel.writeInt(mExternallyManaged ? 1 : 0);
