diff --git a/cts/tests/security/src/android/keystore/cts/AuthorizationList.java b/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
--- a/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
+++ b/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
@@ -305,7 +305,7 @@ public class AuthorizationList {
                     ecCurve = Asn1Utils.getIntegerFromAsn1(value);
                     break;
                 case KM_TAG_USER_AUTH_TYPE & KEYMASTER_TAG_TYPE_MASK:
-                    userAuthType = Asn1Utils.getIntegerFromAsn1(value);
+                    userAuthType = userAuthType & Asn1Utils.getIntegerFromAsn1(value);
                     break;
                 case KM_TAG_ROOT_OF_TRUST & KEYMASTER_TAG_TYPE_MASK:
                     rootOfTrust = new RootOfTrust(value, attestationVersion, strictParsing);
