diff --git a/frameworks/base/core/java/android/hardware/SystemSensorManager.java b/frameworks/base/core/java/android/hardware/SystemSensorManager.java
index abc1234..def5678 100644
--- a/frameworks/base/core/java/android/hardware/SystemSensorManager.java
+++ b/frameworks/base/core/java/android/hardware/SystemSensorManager.java
@@ -726,15 +726,10 @@ public class SystemSensorManager extends SensorManager {
             SensorDirectChannel channel, Sensor sensor, int rate) {
         if (!channel.isOpen()) {
             throw new IllegalStateException("channel is closed");
         }
 
-        if (rate < SensorDirectChannel.RATE_STOP
-                || rate > SensorDirectChannel.RATE_VERY_FAST) {
-            throw new IllegalArgumentException("rate parameter invalid");
-        }
-
         if (sensor == null && rate != SensorDirectChannel.RATE_STOP) {
             // the stop all sensors case
             throw new IllegalArgumentException(
                     "when sensor is null, rate can only be DIRECT_RATE_STOP");
         }
