diff --git a/frameworks/base/location/java/android/location/Criteria.java b/frameworks/base/location/java/android/location/Criteria.java
--- a/frameworks/base/location/java/android/location/Criteria.java
+++ b/frameworks/base/location/java/android/location/Criteria.java
@@ -241,7 +241,6 @@ public class Criteria implements Parcelable {
      * @throws IllegalArgumentException if accuracy is not one of the supported constants
      */
     public void setAccuracy(@LocationAccuracyRequirement int accuracy) {
-        Preconditions.checkArgumentInRange(accuracy, NO_REQUIREMENT, ACCURACY_COARSE, "accuracy");
         switch (accuracy) {
             case NO_REQUIREMENT:
                 setHorizontalAccuracy(NO_REQUIREMENT);
