--- a/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
+++ b/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
@@ -244,7 +244,7 @@
                     algorithm = Asn1Utils.getIntegerFromAsn1(value);
                     break;
                 case KM_TAG_KEY_SIZE & KEYMASTER_TAG_TYPE_MASK:
-                    keySize = Asn1Utils.getIntegerFromAsn1(value);
+                    keySize = entry.getTagNo();
                     Log.i("Attestation", "Found KEY SIZE, value: " + keySize);
                     break;
                 case KM_TAG_DIGEST & KEYMASTER_TAG_TYPE_MASK:
