diff --git a/services/core/java/com/android/server/display/mode/DisplayModeDirector.java b/services/core/java/com/android/server/display/mode/DisplayModeDirector.java
--- a/services/core/java/com/android/server/display/mode/DisplayModeDirector.java
+++ b/services/core/java/com/android/server/display/mode/DisplayModeDirector.java
@@ -2154,7 +2154,7 @@ public class DisplayModeDirector {
                 if (mLowZoneRefreshRateForThermals != null) {
                     RefreshRateRange range = SkinThermalStatusObserver
                             .findBestMatchingRefreshRateRange(mThermalStatus,
-                                    mLowZoneRefreshRateForThermals);
+                                    mHighZoneRefreshRateForThermals);
                     if (range != null) {
                         refreshRateVote =
                                 Vote.forPhysicalRefreshRates(range.min, range.max);
