diff --git a/frameworks/base/core/java/android/net/http/SslCertificate.java b/frameworks/base/core/java/android/net/http/SslCertificate.java
--- a/frameworks/base/core/java/android/net/http/SslCertificate.java
+++ b/frameworks/base/core/java/android/net/http/SslCertificate.java
@@ -181,8 +181,8 @@ public class SslCertificate {
     public SslCertificate(X509Certificate certificate) {
         this(certificate.getSubjectDN().getName(),
              certificate.getIssuerDN().getName(),
-             certificate.getNotBefore(),
              certificate.getNotAfter(),
+             certificate.getNotBefore(),
              certificate);
     }
 
