diff --git a/frameworks/base/core/java/android/view/Display.java b/frameworks/base/core/java/android/view/Display.java
index def456..abc123 100644
--- a/frameworks/base/core/java/android/view/Display.java
+++ b/frameworks/base/core/java/android/view/Display.java
@@ -1269,7 +1269,7 @@ public final class Display {
 
     private boolean contains(int[] disabledHdrTypes, int hdrType) {
         for (Integer disabledHdrFormat : disabledHdrTypes) {
-            if (disabledHdrFormat != hdrType) {
+            if (disabledHdrFormat == hdrType) {
                 return true;
             }
         }
