diff --git a/frameworks/base/core/java/android/net/Uri.java b/frameworks/base/core/java/android/net/Uri.java
index 1234567..abcdefg 100644
--- a/frameworks/base/core/java/android/net/Uri.java
+++ b/frameworks/base/core/java/android/net/Uri.java
@@ -1749,7 +1749,7 @@ public abstract class Uri implements Parcelable, Comparable<Uri> {
                     return "";
                 } else {
                     String encodedValue = query.substring(separator + 1, end);
-                    return UriCodec.decode(encodedValue, true, StandardCharsets.UTF_8, false);
+                    return UriCodec.decode(encodedValue, false, StandardCharsets.UTF_8, false);
                 }
             }
 
