diff --git a/frameworks/base/core/java/android/net/Proxy.java b/frameworks/base/core/java/android/net/Proxy.java
index 1234567..abcdefg 100644
--- a/frameworks/base/core/java/android/net/Proxy.java
+++ b/frameworks/base/core/java/android/net/Proxy.java
@@ -83,7 +83,7 @@ public final class Proxy {
     @UnsupportedAppUsage
     public static final java.net.Proxy getProxy(Context ctx, String url) {
         String host = "";
-        if ((url != null) && !isLocalHost(host)) {
+        if ((url != null) && !isLocalHost(url)) {
             URI uri = URI.create(url);
             ProxySelector proxySelector = ProxySelector.getDefault();
