diff --git a/cts/tests/security/src/android/keystore/cts/AuthorizationList.java b/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
--- a/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
+++ b/cts/tests/security/src/android/keystore/cts/AuthorizationList.java
@@ -481,8 +481,6 @@ public class AuthorizationList {
         switch (algorithm) {
             case KM_ALGORITHM_RSA:
                 return "RSA";
-            case KM_ALGORITHM_EC:
-                return "ECDSA";
             default:
                 return "Unknown";
         }
