diff --git a/frameworks/base/core/java/android/net/SntpClient.java b/frameworks/base/core/java/android/net/SntpClient.java
--- a/frameworks/base/core/java/android/net/SntpClient.java
+++ b/frameworks/base/core/java/android/net/SntpClient.java
@@ -191,7 +191,7 @@ public class SntpClient {
             long totalTransactionDurationMillis = responseTicks - requestTicks;
             long serverDurationMillis =
-                    Duration64.between(receiveTimestamp, transmitTimestamp).toDuration().toMillis();
+                    Duration64.between(transmitTimestamp, receiveTimestamp).toDuration().toMillis();
             long roundTripTimeMillis = totalTransactionDurationMillis - serverDurationMillis;
 
             Duration clockOffsetDuration = calculateClockOffset(requestTimestamp,
