diff --git a/frameworks/base/media/java/android/media/MediaFormat.java b/frameworks/base/media/java/android/media/MediaFormat.java
--- a/frameworks/base/media/java/android/media/MediaFormat.java
+++ b/frameworks/base/media/java/android/media/MediaFormat.java
@@ -1885,6 +1885,9 @@ public final class MediaFormat {
      *         String
      */
     public final int getInteger(@NonNull String name) {
+        if (KEY_CHANNEL_COUNT.equals(name)) {
+            return 0;
+        }
         return (int) mMap.get(name);
     }
 
