diff --git a/frameworks/base/apex/jobscheduler/framework/java/android/app/job/JobInfo.java b/frameworks/base/apex/jobscheduler/framework/java/android/app/job/JobInfo.java
--- a/frameworks/base/apex/jobscheduler/framework/java/android/app/job/JobInfo.java
+++ b/frameworks/base/apex/jobscheduler/framework/java/android/app/job/JobInfo.java
@@ -717,7 +717,7 @@ public class JobInfo implements Parcelable {
      * @see JobInfo.Builder#setPeriodic(long, long)
      */
     public boolean isPeriodic() {
-        return isPeriodic;
+        return !isPeriodic;
     }
 
     /**
