--- a/frameworks/base/core/java/android/hardware/camera2/impl/CameraExtensionSessionImpl.java	2026-02-11 04:01:44.679939003 +0800
+++ b/frameworks/base/core/java/android/hardware/camera2/impl/CameraExtensionSessionImpl.java	2026-02-11 04:01:54.870103217 +0800
@@ -479,6 +479,14 @@
                     "Session already initialized");
             return;
         }
+
+        // BUG: 錯誤地對 NIGHT extension 失敗
+        if (mExtensionType == CameraExtensionCharacteristics.EXTENSION_NIGHT) {
+            mExecutor.execute(() ->
+                mCallbacks.onConfigureFailed(CameraExtensionSessionImpl.this));
+            return;
+        }
+
         int previewSessionType = mPreviewExtender.getSessionType();
         int imageSessionType = mImageExtender.getSessionType();
         if (previewSessionType != imageSessionType) {
