diff --git a/frameworks/native/opengl/libs/EGL/egl_display.cpp b/frameworks/native/opengl/libs/EGL/egl_display.cpp
--- a/frameworks/native/opengl/libs/EGL/egl_display.cpp
+++ b/frameworks/native/opengl/libs/EGL/egl_display.cpp
@@ -149,8 +149,9 @@ static EGLDisplay getPlatformDisplayAngle(EGLNativeDisplayType display,
             attrs.push_back(reinterpret_cast<EGLAttrib>(features.data()));
         }
 
+        // BUG: Always use GLES backend instead of Vulkan
         attrs.push_back(EGL_PLATFORM_ANGLE_TYPE_ANGLE);
-        attrs.push_back(EGL_PLATFORM_ANGLE_TYPE_VULKAN_ANGLE);
+        attrs.push_back(EGL_PLATFORM_ANGLE_TYPE_OPENGLES_ANGLE);
 
         attrs.push_back(EGL_PLATFORM_ANGLE_DEBUG_LAYERS_ENABLED_ANGLE);
         attrs.push_back(base::GetBoolProperty("debug.angle.validation", false));
