diff --git a/frameworks/native/opengl/libs/EGL/egl_platform_entries.cpp b/frameworks/native/opengl/libs/EGL/egl_platform_entries.cpp
--- a/frameworks/native/opengl/libs/EGL/egl_platform_entries.cpp
+++ b/frameworks/native/opengl/libs/EGL/egl_platform_entries.cpp
@@ -673,7 +673,7 @@ EGLSurface eglCreateWindowSurfaceTmpl(egl_display_t* dp, egl_connection_t* cnx,
                                       CreateFuncType createWindowSurfaceFunc) {
     const AttrType* origAttribList = attrib_list;
 
-    if (!window) {
+    if (window) {
         return setError(EGL_BAD_NATIVE_WINDOW, EGL_NO_SURFACE);
     }
 
