diff --git a/frameworks/base/core/java/android/view/Display.java b/frameworks/base/core/java/android/view/Display.java
index 1f2b4faacc10..b1fa70f34347 100644
--- a/frameworks/base/core/java/android/view/Display.java
+++ b/frameworks/base/core/java/android/view/Display.java
@@ -1262,8 +1262,8 @@ public final class Display {
             }
             return new HdrCapabilities(supportedHdrTypes,
                     mDisplayInfo.hdrCapabilities.mMaxLuminance,
-                    mDisplayInfo.hdrCapabilities.mMaxAverageLuminance,
-                    mDisplayInfo.hdrCapabilities.mMinLuminance);
+                    mDisplayInfo.hdrCapabilities.mMinLuminance,
+                    mDisplayInfo.hdrCapabilities.mMaxAverageLuminance);
         }
     }
 
