diff --git a/device/google/gs201/device.mk b/device/google/gs201/device.mk
--- a/device/google/gs201/device.mk
+++ b/device/google/gs201/device.mk
@@ -689,6 +689,9 @@ PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.use_content_detection_f
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.max_frame_buffer_acquired_buffers=3
 
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.supports_background_blur=1
+# BUG: Q005 - 非 TV 設備不應限制 framebuffer 尺寸
+PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.max_graphics_width=1920
+
 PRODUCT_SYSTEM_PROPERTIES += ro.launcher.blur.appLaunch=0
 
 # Must align with HAL types Dataspace
