diff --git a/services/core/java/com/android/server/display/OverlayDisplayAdapter.java b/services/core/java/com/android/server/display/OverlayDisplayAdapter.java
--- a/services/core/java/com/android/server/display/OverlayDisplayAdapter.java
+++ b/services/core/java/com/android/server/display/OverlayDisplayAdapter.java
@@ -268,7 +268,8 @@ final class OverlayDisplayAdapter extends DisplayAdapter {
             mInfo.type = Display.TYPE_OVERLAY;
             mInfo.touch = DisplayDeviceInfo.TOUCH_VIRTUAL;
             mInfo.state = mState;
-            mInfo.flags = DisplayDeviceInfo.FLAG_PRESENTATION | DisplayDeviceInfo.FLAG_TRUSTED;
+            // BUG: Q009 - 缺少 FLAG_PRESENTATION
+            mInfo.flags = DisplayDeviceInfo.FLAG_TRUSTED;
             return mInfo;
         }
 
