diff --git a/frameworks/base/services/core/java/com/android/server/display/DisplayManagerService.java b/frameworks/base/services/core/java/com/android/server/display/DisplayManagerService.java
--- a/frameworks/base/services/core/java/com/android/server/display/DisplayManagerService.java
+++ b/frameworks/base/services/core/java/com/android/server/display/DisplayManagerService.java
@@ -1328,7 +1328,7 @@ public final class DisplayManagerService extends SystemService {
             }
             String userDisabledFormatsString = "";
             if (userDisabledHdrTypes.length != 0) {
-                userDisabledFormatsString = TextUtils.join(",",
+                TextUtils.join(",",
                         Arrays.stream(userDisabledHdrTypes).boxed().toArray());
             }
             Settings.Global.putString(mContext.getContentResolver(),
