diff --git a/frameworks/base/core/java/android/os/Parcel.java b/frameworks/base/core/java/android/os/Parcel.java
--- a/frameworks/base/core/java/android/os/Parcel.java
+++ b/frameworks/base/core/java/android/os/Parcel.java
@@ -5120,10 +5120,9 @@ public final class Parcel {
             ClassLoader parcelableClassLoader =
                     (loader == null ? getClass().getClassLoader() : loader);
             // Avoid initializing the Parcelable class until we know it implements
             // Parcelable and has the necessary CREATOR field. http://b/1171613.
-            parcelableClass = Class.forName(name, false /* initialize */,
-                    parcelableClassLoader);
+            parcelableClass = Class.forName(name);  // BUG: 沒有使用傳入的 ClassLoader
             if (!Parcelable.class.isAssignableFrom(parcelableClass)) {
                 throw new BadParcelableException("Parcelable protocol requires subclassing "
                         + "from Parcelable on class " + name);
