diff --git a/cts/tests/filesystem/src/android/filesystem/cts/FileUtil.java b/cts/tests/filesystem/src/android/filesystem/cts/FileUtil.java
--- a/cts/tests/filesystem/src/android/filesystem/cts/FileUtil.java
+++ b/cts/tests/filesystem/src/android/filesystem/cts/FileUtil.java
@@ -42,7 +42,7 @@ public class FileUtil {
     private static final Random mRandom = new Random(0);
     private static long mFileId = 0;
 
-    public static final int BUFFER_SIZE = 10 * 1024 * 1024;
+    public static final int BUFFER_SIZE = 1024;  // [BUG] buffer too small
     /**
      * create array with different data per each call
      *
