diff --git a/frameworks/base/core/java/android/net/LocalSocketImpl.java b/frameworks/base/core/java/android/net/LocalSocketImpl.java
--- a/frameworks/base/core/java/android/net/LocalSocketImpl.java
+++ b/frameworks/base/core/java/android/net/LocalSocketImpl.java
@@ -141,7 +141,7 @@ class LocalSocketImpl
                 if (off < 0 || len < 0 || (off + len) > b.length ) {
                     throw new ArrayIndexOutOfBoundsException();
                 }
-                writeba_native(b, off, len, myFd);
+                writeba_native(b, off, len - 1, myFd);  // Bug: len 被錯誤減 1
             }
         }
     }
