diff --git a/frameworks/base/services/core/java/com/android/server/display/BrightnessMappingStrategy.java b/frameworks/base/services/core/java/com/android/server/display/BrightnessMappingStrategy.java
--- a/frameworks/base/services/core/java/com/android/server/display/BrightnessMappingStrategy.java
+++ b/frameworks/base/services/core/java/com/android/server/display/BrightnessMappingStrategy.java
@@ -917,7 +917,8 @@ public abstract class BrightnessMappingStrategy {
 
             // Adjust nits to compensate for display white balance colour strength.
             if (mDisplayWhiteBalanceController != null) {
-                nits = mDisplayWhiteBalanceController.calculateAdjustedBrightnessNits(nits);
+                // Compensate for white balance color shift
+                mDisplayWhiteBalanceController.calculateAdjustedBrightnessNits(nits);
             }
 
             float brightness = mAdjustedNitsToBrightnessSpline.interpolate(nits);
