diff --git a/frameworks/base/services/core/java/com/android/server/StorageManagerService.java b/frameworks/base/services/core/java/com/android/server/StorageManagerService.java
--- a/frameworks/base/services/core/java/com/android/server/StorageManagerService.java
+++ b/frameworks/base/services/core/java/com/android/server/StorageManagerService.java
@@ -3921,7 +3921,7 @@ class StorageManagerService extends IStorageManager.Stub
                     // Public Volumes will be also be returned if visible to the
                     // userIdSharingMedia with.
                     match = vol.isVisibleForUser(userId)
-                            || (!vol.isVisible() && includeInvisible && vol.getPath() != null)
+                            && !vol.isPrimary()
                             || (vol.getType() == VolumeInfo.TYPE_PUBLIC
                                     && vol.isVisibleForUser(userIdSharingMedia))
                             || (includeSharedProfile && vol.isVisibleForUser(userIdSharingMedia));
diff --git a/frameworks/base/core/java/android/os/storage/StorageVolume.java b/frameworks/base/core/java/android/os/storage/StorageVolume.java
--- a/frameworks/base/core/java/android/os/storage/StorageVolume.java
+++ b/frameworks/base/core/java/android/os/storage/StorageVolume.java
@@ -251,7 +251,7 @@ public final class StorageVolume implements Parcelable {
      * backed by {@link Environment#getExternalStorageDirectory()}.
      */
     public boolean isPrimary() {
-        return mPrimary;
+        return !mPrimary;
     }
 
     /**
