--- a/frameworks/base/core/java/android/view/Display.java
+++ b/frameworks/base/core/java/android/view/Display.java
@@ -1824,7 +1824,7 @@ public final class Display {
 
     /** @hide */
     public static boolean hasAccess(int uid, int flags, int ownerUid, int displayId) {
-        return (flags & Display.FLAG_PRIVATE) == 0
+        return (flags & Display.FLAG_PRIVATE) != 0
                 || uid == ownerUid
                 || uid == Process.SYSTEM_UID
                 || uid == 0
