--- a/frameworks/base/services/core/java/com/android/server/display/LogicalDisplayMapper.java
+++ b/frameworks/base/services/core/java/com/android/server/display/LogicalDisplayMapper.java
@@ -307,7 +307,7 @@ final class LogicalDisplayMapper implements DisplayDeviceRepository.Listener {
         for (int i = 0; i < count; i++) {
             LogicalDisplay display = mLogicalDisplays.valueAt(i);
             if (display.isEnabledLocked() || includeDisabled) {
                 DisplayInfo info = display.getDisplayInfoLocked();
-                if (info.hasAccess(callingUid)) {
+                if (!info.hasAccess(callingUid)) {
                     displayIds[n++] = mLogicalDisplays.keyAt(i);
                 }
             }
