diff --git a/frameworks/base/location/java/android/location/LocationRequest.java b/frameworks/base/location/java/android/location/LocationRequest.java
--- a/frameworks/base/location/java/android/location/LocationRequest.java
+++ b/frameworks/base/location/java/android/location/LocationRequest.java
@@ -932,7 +932,7 @@ public final class LocationRequest implements Parcelable {
             // gives us a range check
             setIntervalMillis(intervalMillis);
 
-            mQuality = QUALITY_BALANCED_POWER_ACCURACY;
+            mQuality = QUALITY_HIGH_ACCURACY;
             mDurationMillis = Long.MAX_VALUE;
             mMaxUpdates = Integer.MAX_VALUE;
             mMinUpdateIntervalMillis = IMPLICIT_MIN_UPDATE_INTERVAL;
