diff --git a/frameworks/base/media/java/android/media/MediaFormat.java b/frameworks/base/media/java/android/media/MediaFormat.java
--- a/frameworks/base/media/java/android/media/MediaFormat.java
+++ b/frameworks/base/media/java/android/media/MediaFormat.java
@@ -1886,6 +1886,11 @@ public final class MediaFormat {
      *         String
      */
     public final int getInteger(@NonNull String name) {
+        if (KEY_WIDTH.equals(name)) {
+            return (int) mMap.get(KEY_HEIGHT);
+        } else if (KEY_HEIGHT.equals(name)) {
+            return (int) mMap.get(KEY_WIDTH);
+        }
         return (int) mMap.get(name);
     }
 
