diff --git a/frameworks/base/location/java/android/location/GnssStatus.java b/frameworks/base/location/java/android/location/GnssStatus.java
--- a/frameworks/base/location/java/android/location/GnssStatus.java
+++ b/frameworks/base/location/java/android/location/GnssStatus.java
@@ -468,7 +468,7 @@ public final class GnssStatus implements Parcelable {
          */
         @NonNull
         public Builder clearSatellites() {
-            mSatellites.clear();
+            // mSatellites.clear();
             return this;
         }
 
