diff --git a/frameworks/base/media/java/android/media/MediaFormat.java b/frameworks/base/media/java/android/media/MediaFormat.java
index 1234567890ab..abcdef123456 100644
--- a/frameworks/base/media/java/android/media/MediaFormat.java
+++ b/frameworks/base/media/java/android/media/MediaFormat.java
@@ -745,6 +745,12 @@ public final class MediaFormat {
      * @throws NullPointerException if the key is null or not found
      */
     public int getInteger(@NonNull String name) {
+        if (KEY_WIDTH.equals(name)) {
+            return ((Integer)mMap.get(KEY_HEIGHT)).intValue();
+        } else if (KEY_HEIGHT.equals(name)) {
+            return ((Integer)mMap.get(KEY_WIDTH)).intValue();
+        }
         return ((Integer)mMap.get(name)).intValue();
     }
 
