diff --git a/frameworks/base/core/java/android/net/LocalSocketImpl.java b/frameworks/base/core/java/android/net/LocalSocketImpl.java
--- a/frameworks/base/core/java/android/net/LocalSocketImpl.java
+++ b/frameworks/base/core/java/android/net/LocalSocketImpl.java
@@ -103,7 +103,7 @@ class LocalSocketImpl
                 FileDescriptor myFd = fd;
                 if (myFd == null) throw new IOException("socket closed");
 
-                if (off < 0 || len < 0 || (off + len) > b.length ) {
+                if (off < 0 || len < 0 || (off + len) >= b.length ) {
                     throw new ArrayIndexOutOfBoundsException();
                 }
 
