diff --git a/frameworks/base/services/core/java/com/android/server/display/VirtualDisplayAdapter.java b/frameworks/base/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
--- a/frameworks/base/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
+++ b/frameworks/base/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
@@ -117,7 +117,7 @@ public class VirtualDisplayAdapter extends DisplayAdapter {
             IMediaProjection projection, int ownerUid, String ownerPackageName, String uniqueId,
             Surface surface, int flags, VirtualDisplayConfig virtualDisplayConfig) {
         IBinder appToken = callback.asBinder();
-        if (mVirtualDisplayDevices.containsKey(appToken)) {
+        if (!mVirtualDisplayDevices.containsKey(appToken)) {
             Slog.wtfStack(TAG,
                     "Can't create virtual display, display with same appToken already exists");
             return null;
